### PR TITLE
in_tcp: Improve the performance for multiple events case

### PR DESCRIPTION
**Pull request by repeatedly**
**Which issue(s) this PR fixes**: 
None

**What this PR does / why we need it**: 
Use `MultiEventStream` to improve the performance for multiple events in receive data.

**Docs Changes**:
No need.

**Release Note**: 
Same as title

### CODE REVIEW

File: lib/fluent/plugin/in_tcp.rb [hunk 1]
```diff
@@ -83,7 +85,32 @@ def start
               router.emit(tag, time, record)
             end
           end
-          conn.buffer.slice!(0, pos) if pos > 0
+          buf.slice!(0, pos) if pos > 0
+        end
+      else
+        server_create(:in_tcp_server_batch_emit, @port, bind: @bind, resolve_name: !!@source_hostname_key) do |data, conn|
+          conn.buffer << data
```
Comment (ganmacs) [inline lib/fluent/plugin/in_tcp.rb]: Is it needed if using a local variable(`buf`)? #buffer is just a string. It is used only for buffering https://github.com/fluent/fluentd/blob/5a7d94941371dddcb7150a73ddbe142102a68964/lib/fluent/plugin_helper/server.rb#L448.

Comment (repeatedly) [reply lib/fluent/plugin/in_tcp.rb]: `buf = conn.buffer` is for avoiding `buffer` accessor call in `while`.

Comment (ganmacs) [reply lib/fluent/plugin/in_tcp.rb]: My question is `conn.buffer << data` is needed? `conn.buffer` will not be used in the following logic.

Comment (repeatedly) [reply lib/fluent/plugin/in_tcp.rb]: Yes. `conn.buffer` may have previous data so need to concat before use.

Comment (ganmacs) [reply lib/fluent/plugin/in_tcp.rb]: ah, I see 👍

File: lib/fluent/plugin/in_tcp.rb [hunk 2]
```diff
@@ -83,7 +85,32 @@ def start
...
+                next
+              end
+
+              time ||= extract_time_from_record(record) || Fluent::EventTime.now
+              record[@source_address_key] = conn.remote_addr if @source_address_key
+              record[@source_hostname_key] = conn.remote_host if @source_hostname_key
+              es.add(time, record)
+            end
+          end
+          router.emit_stream(@tag, es)
```
Comment (ganmacs) [inline lib/fluent/plugin/in_tcp.rb]: This PR has already been merged. but I found a considering point.

if `es` becomes bigger than [Buffer#chunk_size_over?](https://github.com/fluent/fluentd/blob/2ad5b6de80cdd64d2ba27526f806e91323e12a68/lib/fluent/plugin/buffer.rb#L564), fluentd shows warning log and does rollback constantly.
so I think it needs to add a configuration parameter to set the size of stream or something.

Comment (ganmacs) [reply lib/fluent/plugin/in_tcp.rb]: > so I think it needs to add a configuration parameter to set the size of streams or something.

re-think carefully, In this situation,  we can set`chunk_limit_size`. so it's not problem.

**Pull request merged**